pkg/repositories/database.go

```diff
@@ -116,17 +115,17 @@ func createPostgresDbIfNotExists(ctx context.Context, gormConfig *gorm.Config, p
 		return gormDb, nil
 	}
 
-	// Check if its invalid db code error
-	cErr, ok := err.(repoErrors.ConnectError)
-	if !ok {
-		logger.Errorf(ctx, "Failed to cast error of type: %v, err: %v", reflect.TypeOf(err),
-			err)
+	// if it is an invalid db code error, continue, otherwise return
+	pgErr := &pgconn.PgError{}
+	if !errors.As(err, &pgErr) {
```

Comment:
Nit: Can you add a few unit tests to cover the newly added lines

Reply:
It's pretty hard to write unit tests without refactoring all of this, because this function potentially opens multiple connections to Postgres directly in its code.

To do this, I'm thinking of creating a DialectorBuilder function/struct that takes a DSN and returns a Dialector, and pass it to `createPostgresDbIfNotExists` as an arg. DialectorBuilder would be created in the `GetDB` function (or further up). I would also need to create a mock DialectorBuilder which will return a mock Dialector that returns the configured errors.

Alternatively, I could just extract these few error handling lines of code and write some tests for it, but that feels like breaking up the code just for the sake of writing tests for specific lines, when the whole function should be tested.

(Ideally, I don't think flyteadmin should even have the permission to `CREATE DATABASE` - that should be a db admin only permission. But this is a much bigger change across the board.) 

Looking at the existing tests, I don't think there's any coverage here at the moment.

@pmahindrakar-oss any thoughts or other ideas? Would you like me to refactor this to include unit tests?

Reply:
Ok sounds good. Thanks @slai